lib/auth/routes_authtype.js

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2015-2018 _floragunn_ GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+/*
+ * Portions Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+module.exports = function (pluginRoot, server, kbnServer, APP_ROOT, API_ROOT) {
+
+  /**
+   * Auth type API that returns current auth type configured on Kibana Server.
+   *
+   * GET /_plugin/kibana/_opendistro/api/authtype
+   * Response:
+   *  200 OK
+   *  {
+   *    authtype: saml
+   *  }
+   */
+   server.route({
+       method: 'GET',
+       path: `/_plugin/kibana/_opendistro/api/authtype`,
+       handler: (request, h) => {
+           return {
+              authtype: server.config().get('opendistro_security.auth.type')
```

Comment:
if I remember it correctly, `auth.type` could be empty `''`, which means default (basicauth) in current implementation. can you verify it? if so, maybe you want to return `basicauth` when this config value is empty.

https://github.com/opendistro-for-elasticsearch/security-kibana-plugin/blob/master/index.js#L42

Reply:
Yes, thanks for pointing this out. I will update the code.